### DISCUSSION TITLE
Ayuda con los ejercicios de VUE

### DISCUSSION THREAD

**Discussion by JossPerezT** [LaunchX-InnovaccionVirtual/FrontEnd-Mision#561] [Q&A] [answered]
Hola explores. 
Espero me puedan ayudar, ando atrasada con la última práctica. 
Estoy haciendo los ejercicios de la semana de VUE, instale node y ejecute `npm install -g @vue/cli`
![gitbash](https://user-images.githubusercontent.com/84040594/160764686-4d8d1aaa-ca39-438d-acb2-e0b6deab50fb.PNG)

Pero al momento de crear un archivo VUE no me lo permite:
Por medio del comando que nos dio el MC `vue create launch` solo queda el mensaje de que esta creando pero ya no hace nada más.
![vue create](https://user-images.githubusercontent.com/84040594/160764952-b52f1533-2896-43fd-9498-b79bd5e612ef.PNG)
Y también intenté hacerlo por los comandos que recomendaron en otras discusisions y tampoco me lo permite y no entendí lo de `mkdir` 
![npm init](https://user-images.githubusercontent.com/84040594/160765134-9b178200-67dd-4ded-b7bc-ca131c77b7d7.PNG)

En algún punto me creo un package-lock.json que es lo que muestra las capturas.

**Comment by RodolfoMorquecho** [comment]
No se si ya lo resolviste pero en mi caso me aparecía que no me reconocía el comando vue hasta que cerré y abrí VSC 2 veces, por cierto lo hiciste desde el command prompt?

**Reply by JossPerezT** [reply]
Si, lo dice desde el command prompt y ya lo cerré varias veces. Lo que si no he intentado es reiniciar la computadora 🤔

**Comment by iRetr0o** [comment]
Al momento de crearlo te aparece un error o algo después de un tiempo?

Yo pienso que porque son demasiados archivos los que está creando es la razón por la que se tarda un poco.

**Reply by JossPerezT** [reply]
No me aparece nada de errores. Lo deje toda la noche creyendo lo mismo y no paso nada más

**Comment by MarcosSilvaG** [comment]
Hola!!! intenta renombrar tu carpeta y sustituir los espacios por - o _ a veces dichos espacios generan errores

**Reply by JossPerezT** [reply]
Ya cambié los nombres pero sigue igual

**Comment by oscardanielM** [comment]
Por lo que veo no te esta contruyendo bien tu proyecto, podrias generar otra carpeta ejecutar el codigo para crearla con vue3

**Reply by JossPerezT** [reply]
Ya hice una nueva carpeta y arroja el mismo mensaje. Será que lo dejo directo en el escritorio?

**Reply by oscardanielM** [reply]
No creo que sea eso, podrias instalar todo otravez, por lo que yo note al instalar el note no me descargo correctamente npn y tuve que buscar por otro lado

**Reply by JossPerezT** [reply]
Reinstale todo y luego una versión anterior y tampoco me deja.
Y cheque la versión de npm y se supone que si está bien instalada 🤔

**Reply by oscardanielM** [reply]
Jaja veo muy complejo tu caso jaja investigare un poco

**Reply by JossPerezT** [reply]
Jaja ya no sé que hacer, mi laptop no quiere que haga esta práctica 🤦🏽‍♀️😂

**Reply by oscardanielM** [reply]
ja opino que te vallas por las versiones mas recientes y que primero instales note y revises si te instalo el npm correctamente de lo contrario tendrias que istalarlo, despues los comandos de vue ja crees una carpeta normalita asi solita ja y dentro de esa corrar el vue create, si lo haces desde visual en la powershell te marcara error es mejor que lo hagas con la terminal de command y ahora si ja deberia dejarte asi chido jaja

**Reply by JossPerezT** [reply]
Haré todo de nuevo y seguiré lo que me comentas. Espero esta vez quedé. 
Muchas gracias!!

**Reply by oscardanielM** [reply]
Vale cualquier cosa pues ya nos notificas

**Comment by david-malpica** [comment]
Instalaste el CLI desde el CMD?

**Reply by JossPerezT** [reply]
Desde Git Bash. Y verifique la version en el CMD.

**Reply by david-malpica** [reply]
Tal vez el Bash omitió algo en la instalación instala el CLI desde CMD, y reinicias tu computadora/laptop, después instala la extensión de [Vetur](https://marketplace.visualstudio.com/items?itemName=octref.vetur), esperando sea la solucion

**Reply by JossPerezT** [reply]
Ya lo instale en el CMD y según aparece está bien instalado porque aparecen bien las versiones.

**Reply by david-malpica** [reply]
- Instalaste la versión LTS de node?
- Instalaste el CLI como administrador?
- Revisaste el firewall?

**Reply by JossPerezT** [reply]
Si. 
Si.
Eso sí no lo he checado, lo haré a lo mejor es eso.

**Comment by Ingrzs** [comment]
Asi me paso a mi la primera ves demoro y no pasaba nada, la cuestión era mi internet , desconecte y volví a conectar y listo.

**Reply by JossPerezT** [reply]
Ya lo intenté y cambié de red pero tampoco, a lo mejor aún necesita más tiempo.

**Comment by ismagician** [comment]
¿ Al crear el proyecto te crea la carpeta de este? Porque si no intenta crear la carpeta. El mkdir es un comando que sirve para crear una carpeta.

**Reply by JossPerezT** [reply]
Cree una nueva carpeta y cambió a que no encuentra el directorio.
![image](https://user-images.githubusercontent.com/84040594/160972257-dae1ca5e-8e94-4ae2-b6fe-0e0e147c111c.png)
Ingresé al directorio y vuelve a mandar mkdir 
![image](https://user-images.githubusercontent.com/84040594/160972337-ac2d1f5c-742a-4e73-882d-a6b2dbeddb5a.png)

**Reply by ismagician** [reply]
Intenta con 
![imagen](https://user-images.githubusercontent.com/92187317/160975090-3f73ceb1-d574-4418-a7b8-479b698ef45d.png)
o instala vue cli con yarn
https://cli.vuejs.org/guide/installation.html

**Reply by JossPerezT** [reply]
Ni me reconoce `yarn` como comando, solo npm 🤔

**Reply by ismagician** [reply]
debes instalar yarn
https://classic.yarnpkg.com/lang/en/docs/install/#windows-stable

**Reply by JossPerezT** [reply]
Va, lo intentaré. Gracias!!!!!

**Reply by JossPerezT** [reply]
Hola. 
Instale yarn 
![image](https://user-images.githubusercontent.com/84040594/161197282-7246255f-da5f-43ab-853d-27fc80eaa569.png)
![image](https://user-images.githubusercontent.com/84040594/161197379-39c33af8-fb90-4967-bee6-7f26d4d1822a.png)


Y al crear proyecto le dije que usara yarn en lugar de npm. 
Me creó la nueva carpeta y package.json 
![image](https://user-images.githubusercontent.com/84040594/161197610-d91e8f62-32e7-49d6-b3de-2995082b1535.png)

Pero ya se quedó ahí, ya no avanzo más. Seguiré esperando o lo volveré a intentar

**Comment by HugoHerreraR** [comment]
intenta crearlo ejecutando el comando vue ui

**Reply by JossPerezT** [reply]
lo estoy dejando crear el archivo, ojalá así se pueda 🤞🏽

**Reply by HugoHerreraR** [reply]
https://www.codejourney.net/2021/04/how-to-fix-npm-err-enoent-enoent-no-such-file-or-directory-rename/
Encontré esa pagina por si te sirve

**Reply by JossPerezT** [reply]
Anoche la encontré y no le había entendido 😅 

Deje creando el proyecto una hr y solo queda cargando y ya intente lo que dice la página pero no hizo ningún cambio

**Reply by HugoHerreraR** [reply]
Posiblemente haya generado un problema durante la instalación de Vue

**Reply by JossPerezT** [reply]
Supongo que si, ahorita estoy desinstalando y borrando todo de nuevo a ver si se soluciona el problema

**Comment by JossPerezT** [comment]
Otra consulta, a lo mejor ahí está mi problema. 
Al instalar node me da un mensaje de que necesita instalar otras herramientas para su funcionamiento. Las veces que reinstale le puse que si agregara las herramientas automáticamente (algunas como un instalador de visual studio, Python, chocolate, PowerShell, etc. Algunas los instalo por PowerShell).


Mi duda es, **¿ustedes dejaron que se instalarán las herramientas o solo saltaron ese paso y se quedaron solo con node solo? **

**Reply by oscardanielM** [reply]
Ja yo le di instalar todo jaja aunque usando la powershell no me funciono bien, tuve que usar el Command Prompt

**Reply by JossPerezT** [reply]
Ah bueno, entonces ese no es el problema aquí 😅
Gracias!

**Reply by JuanSalvador73** [reply]
Tengo un problema parecido, solo quiero saber si la instalación de VUE cli la hacen desde la terminal de Visual Code Studio o desde la terminal de sus computadoras - en mi caso Window 10.  Veo que varios tenemos un problema similar, hay un paso que estamos omitiendo o pasando  por alto.

**Reply by oscardanielM** [reply]
Yo lo estoy haciendo desde mi consola de windows

**Reply by JossPerezT** [reply]
@JuanSalvador73  según tengo entendido sería lo mismo en las consolas de VSC y windows (si estoy mal corríjanme). Aún así no puedo crear el proyecto en ninguna.
Pero intenta con YARN, un compañero lo dejo aquí en la discusión, a lo mejor te sirve a ti.

**Reply by JossPerezT** [reply]
[]()https://youtu.be/9qWyoKrz-v8
[]()https://youtu.be/NdXQUD88FT0
[]()https://youtu.be/ipmhBYqIP44 



Esos son unos de los vídeos que más he entendido, en mi caso no me sirvieron pero lo mejor te ayudan.

**Reply by JuanSalvador73** [reply]
Gracias Joss, gracias Oscar Daniel. Checaré esos enlaces. Al parecer mi problema es que por alguna razón mi lab no me reconoce VUE, simplemente como que no instala el npm. Le doy vue --version y no lo reconoce.

Seguiré intentando, ya llevo días en esto y horas y horas en estos foros y Google sin saber que pasa con mi lab. Si encuentro la solución la comunicaré por aquí, Gracias nuevamente.

**Reply by ismagician** [reply]
Si le diste a la opción de que te agregara otras herramientas automáticamente, te debió de haber instalado Chocolatey (es un manejador de paquetes) con ese puedes instalar nodejs. Para checar si lo tienes instalado, desde cmd o powershell  pon
 ```choco --version```.
Para instalar  nodejs, (abre terminal como adminstrador) : versión LTS  ```choco install nodejs-lts``` o la versión Current
 ```choco install nodejs.install``` (yo solo le puse ```choco install nodejs``` y me instaló la Current)

**Reply by JossPerezT** [reply]
Hola. Hice lo que me indicaste Isma
![image](https://user-images.githubusercontent.com/84040594/161193338-f54b5582-a585-4f78-82b7-76a67eba9a8b.png)
![image](https://user-images.githubusercontent.com/84040594/161193404-5866536d-3ae6-4587-8790-7b1125fbca81.png)


Me arrojó esos resultados, pero sigue sin crearse el proyecto de VUE

**Reply by JossPerezT** [reply]
Hola @JuanSalvador73, ya pudiste resolver tu problema?

**Reply by JuanSalvador73** [reply]
Hola [JossPerezT](https://github.com/JossPerezT), no, no he podido arreglar el problema, no había ya vuelto a preguntar para no fastidiar, como ya he preguntado bastante - y ya llevo 5 días googleando y nada - pues estoy considerando usar otro software, parece que REACT es lo que otros explorers recomendaban, pero aún no lo he instalado.

Incluso ya probé modificar las variables de entorno en panel de control como lo indica este tuto, pero nada:

https://es.stackoverflow.com/questions/249136/ayuda-al-instalar-vue

**Reply by JuanSalvador73** [reply]
Y comprobé como tu que versión de npm, node y choco tienen instalados mi lab.  Pero a la hora de querer instalar VUE, y crear el proyecto, la misma leyenda: 

""vue" no se reconoce como un comando interno o externo,
programa o archivo por lotes ejecutable."

¿Y a ti como te ha ido?

**Reply by JossPerezT** [reply]
Igual intentaré con las variables en el panel de control. 
Al instalar y usar Yarn en lugar de npm  sentí como un "avance" porque ya me creaban las carpetas de los proyectos y los package.json, pero se quedaba cargando y no avanzaba. Y luego no se que hice pero volví al inicio, ya no se crean los json 🤦🏽‍♀️

**Reply by ismagician** [reply]
@JuanSalvador73 si pudiste solucionar tu problema? Porque si no checa dentro de la carpeta donde esta npm 
```C:\Users\TuUsuario\AppData\Roaming\npm```
existan archivos relacionados con Vue, te paso una captura de mas o menos como está
![imagen](https://user-images.githubusercontent.com/92187317/161349976-9cb902ce-68b0-447a-a25e-4e7b2f1fd488.png)
![imagen](https://user-images.githubusercontent.com/92187317/161350077-d60a525c-c9a5-45e3-8876-efcf73472874.png)
Si no están los archivos de Vue pues me temo que no lo instaló.
Si están los archivos de Vue, desde cmd (primero ubicate en la carpeta trabajo)  ejecuta el siguiente comando
```C:\Users\TuUsuario\AppData\Roaming\npm\vue --version```  
Si funciona pues crea tu proyecto
```C:\Users\TuUsuario\AppData\Roaming\npm\vue create nombredelproyecto```

**Reply by JuanSalvador73** [reply]
Hola de nuevo Joss. Anoche ví este mensaje, y veo con horror que efectivamente mi lab no está bajando los archivos completos de VUE tal y como se muestran en tus fotos. Así que ahora debo averiguar porque, ya chequé mi firewall y antivirus y no son el problema. 

Al menos ya pude inscribirme a BackEnd aunque mis repositorios están incompletos. Como entré a Java, en el live mencionaron que usaremos VUE para una práctica. Así que debo encontrar la solución.  Pero gracias a tus screenshots al menos ya veo cual es el problema. Seguiré en ello. Gracias por todo y suerte en la misión que escogiste. Saludos.

**Reply by JonathanGuerraGro** [reply]
Hola, @JuanSalvador73, te recomiendo abrir un Discussion propio explicando tu caso con capturas de pantalla, de seguro la comunidad te apoya a resolverlo, esto para centrarse en tu problema y que no se revuelva con este Discussion ya que al parecer lo tuyo es algo diferente, saludos.

**Reply by carlos-gmx3** [reply]
Como dice @JonathanGuerraGro 
No van allegar muchos por acá ya que la discusión está marcada como resuelta

**Reply by JossPerezT** [reply]
Cómo comentan, mejor la discusión aparte y sabremos más a detalle, al menos ya sabemos cuál es el problema y podemos investigar para ayudar. 
Intentaste con el NVM? Se supone es un administrador de versiones y puedes instalar las recientes o las LTS (eso entendí en el vídeo) 
Te vuelvo a mandar la liga
[]()https://youtu.be/9qWyoKrz-v8

**Reply by JuanSalvador73** [reply]
Buenas Explorers. Si, de hecho solo había puesto el post como aviso, si estoy considerando abrir una Discusión para dar a conocer mi problema, aunque dudo que ya lo vean, como ya iniciamos este lúnes el BackEnd pues todos vamos a estar muy atareados con nuevos retos y tareas como para ver post "antiguos", o eso creo yo.

Como sea cualquier cambio lo haré saber con una nueva Discusión,  y nuevamente Gracias Joss, Carlos y Jonathan por sus respuestas. En cuanto tenga algo lo daré a conocer en una Discusión. Saludos y suerte :D

**Reply by JuanSalvador73** [reply]
Buenas, solo para avisarles que ya logré instalar VUE cli en mi labtop. Mas tarde - ahora son las 3 de la mañana y ando cansado - abro una Discusión de mi problema y como lo solucionamos por si a alguien mas le sirve. No he creado proyecto aún, confío no tener mas inconvenientes.  Gracias nuevamente por todo.

**Comment by carlos-gmx3** [comment]
que te devuelve node --version, npm --version y vue --version?

**Reply by JossPerezT** [reply]
De node era la 17.8.0 si no me equivoco y las demás te mentiría, no recuerdo cuáles eras y borre todo para volver a instalar. Pero si me regresaba la versión.

**Reply by JossPerezT** [reply]
choco:
1.0.1

npm 
8.5.5

node 
v17.8.0

vue 
@vue/cli 5.0.4

**Reply by carlos-gmx3** [reply]
Ya intentaste con...
![imagen](https://user-images.githubusercontent.com/99064239/161218897-4e40535d-df2d-4a10-9150-8635403a5902.png)
LTS es Long Term Service
Es más estable para proyectos a largo plazo según vi en un tutorial por ahí
![imagen](https://user-images.githubusercontent.com/99064239/161218951-e28732e7-c9b4-455a-bdaf-e8910b0911da.png)
Igual podrías checar si las variables de entorno de node están configuradas en el path global del equipo y no solo en tu usuario
![imagen](https://user-images.githubusercontent.com/99064239/161220241-bccb976a-cd63-4a03-80ab-0951c53df99d.png)
Por último, tal ves tengas algún tipo de restricción del sistema para crear archivos
En la instalación del windows de mi equipo hice una partición dedicada a temas de programación, la unidad V: que llame playground, aquí hago mis programas, instalo maquinas virtuales etc
![imagen](https://user-images.githubusercontent.com/99064239/161220669-33a1439a-e2cb-49d9-9a97-4b16ad8b0fbc.png)
Si todo lo anterior falla solo quedará ayudarte a checar tu compu con una aplicación de escritorio remoto
Suerte

**Reply by carlos-gmx3** [reply]
Otra cosa, si te das cuenta, cuando creas el proyecto te devuelve un código de error
El 4058
https://stackoverflow.com/questions/64040036/vue-cli-got-error-code-4058-when-create-project
![imagen](https://user-images.githubusercontent.com/99064239/161222493-3ea87e69-9b87-4ecb-bb3f-3ff85508217a.png)

**Reply by JossPerezT** [reply]
Si instale la versión LTS. 

Intentaré con las variables, ojalá sea eso, muchas gracias por el tutorial 🚀

**Comment by AngelCruzO** [comment]
Intentaste crear el proyecto de forma manual, sin darle la opción de default. Solo con Router, Vuex y Babel. Tambien si tienes acceso a otro equipo, trata de instalarlo ahi, ya que cuando desintalas algo, no se pierde por completo, las configuraciones se quedan guardadas y cuando instalas de nuevo, muchas veces retoma esos archivos de configuración. 

Otra cosa es que yo instale el node sin ningun complemento, porque la primera vez que instale con chocolate se tardo mucho tiempo, no instalo bien y tenia errores. Si sigues con errores, puedes ocupar el cdn de Vue, lo malo de esa solución es que tendrias que crear los archivos y carpetas.

**Reply by JossPerezT** [reply]
Ok, voy a instalarlo sin complementos y a ver si así queda. 
Toda la creación la hice con default 3.

**Reply by JossPerezT** [reply]
Intenté borrar todas las herramientas extras que instalar y eliminar los archivos que se crearon como chocolatey de los archivos. Siguió sin crear los archivos.

**Reply by AngelCruzO** [reply]
Pues te vuelvo a comentar si tienes acceso a otro equipo donde lo puedas instalar hazlo, sino cuentas con ese equipo extra y sabes crear Máquinas Virtuales es otra opción para experimentar sin dañar tu equipo principal

**Reply by JossPerezT** [reply]
Va que va. Sigo buscando un equipo. Gracias!!!

**Reply by AngelCruzO** [reply]
Sino investiga sobre Máquinas Virtuales, tal vez sea un poco tardado dependiendo de tu equipo, pero si no funciona la borras y la vuelves a crear. Ya que te funcione Vue la dejas en lo que sacas la práctica.

Otra opción es que ocupes el cdn, se hizo una práctica con eso, lo malo es que tienes que ir creando todo (carpetas, archivos), pero es una buena opción

**Reply by JossPerezT** [reply]
Si, planeaba hacer la máquina con azure, aún tengo credito de la cuenta pasada. 
Con el CDN me salió muy bien el ejercicio pero sentía que CLI era más práctico en el sentido de los archivos pero si no encuentro la solución pronto lo haré en alguna de esas formas. No habrá más remedio

**Reply by AngelCruzO** [reply]
Es que si está muy raro, que no te esté funcionando.

**Reply by JossPerezT** [reply]
Ya hasta me da vergüenza porque pareciera que no hago ningún proceso que me dicen y no queda 😓😅

**Reply by AngelCruzO** [reply]
Suele pasar, yo por ejemplo no me corria el emulador de Android Studio y si me tarde como 2 meses para que me corriera

**Comment by JonathanGuerraGro** [comment, marked as answer]
Hola, estuve leyendo y viendo las imagenes me di cuenta de algo, ¿no sera problema del internet? o de algun firewall - antivirus que te esta bloqueando la conexion?

Pregunto porque aqui mismo te dice que tiene problemas de conexion a la hora de empezar a crear (aunque de yarn no se bien que onda, es una suposicion jaja)
![vue-error3](https://user-images.githubusercontent.com/86798998/161340628-e1493678-e939-4f2d-9ab4-b9321eaa49da.png)


y en la primera imagen que subiste se queda intentando crearlo, hice una prueba quitando mi internet y me pasa algo similar, por eso mi duda de si hay algun problema de conexion o si tienes algun firewall o antivirus que te lo este bloqueando

![vue-error02](https://user-images.githubusercontent.com/86798998/161340652-4987b8af-0778-4171-8a0e-aa0f5b9ad216.png)

**Reply by JossPerezT** [reply]
Ajá, también lo consideré cuando vi ese tipo de mensajes, y cheque la velocidad de internet y tengo entre 25-30 de carga y descarga, ¿Crees que necesite más?

 Igual lo conecté al repetidor de red y sigue cargando, lo dejé toda noche en qué nadie usa la red y solo continúa cargando.

 Al instalar yarn en CMD si me dio un error de conexión pero volví a ejecutar el comando  y lo ejecutó bien, sin ningún mensaje de error.

**Reply by JossPerezT** [reply]
Solo lo que si no he checado es el antivirus, apenas haré la prueba con esa idea 🤔🤔

**Reply by JonathanGuerraGro** [reply]
Con ese internet estable deberia ser suficiente, y si, ya solo probando checar si el antivirus no lo bloquea y de igual manera desactivando los firewalls de windows defender en lo que haces la prueba a ver que tal.

**Reply by JossPerezT** [reply]
Ya quedó, ese fue mi problema Jonathan, muchas gracias por el apoyo! 🚀

**Comment by DanyVeneno** [comment]
Hola JossPerezT  no se si ya pudiste crear tu proyecto pero te dejo un tutorial de udemy muy sencillo talvez te pueda servir
https://www.udemy.com/course/vue-3-js-iniciate-en-este-framework/ es gratis solo debes crear una cuenta en esa plataforma y no es publicidad 
1-https://www.youtube.com/watch?v=_DQP6UxKEn8&ab_channel=jonmircha    este es para la instalación de Nodejs
2-https://www.youtube.com/watch?v=dGK2sayqW9o&list=PLEtcGQaT56ch3jHPeZfakUT3zRm9QfVQi&ab_channel=Jes%C3%BAsConde   y este es otra opción 


3-Otra opcion es utilizar la consola o el powersShell

4- O habilitar un poco de espacio en tu equipo  https://www.youtube.com/watch?v=ywCOs4XyNgg&ab_channel=NJ-Tutoriales

Saludos.

**Reply by JossPerezT** [reply]
Hola! Muchas gracias por los recursos, los checare para ver si se soluciona el problema 🚀

**Reply by carlos-gmx3** [reply]
Aún no queda? La opción nuclear es formatear el equipo

**Reply by AngelCruzO** [reply]
@carlos-gmx3 No exageres, con una VM se puede solucionar

**Reply by JossPerezT** [reply]
Ya quedó @carlos-gmx3, era el antivirus y firewall 😅😅

**Comment by JossPerezT** [comment]
Hola Explores. Muchas gracias a cada uno de ustedes por su tiempo, ideas, tutoriales y toda información que me proporcionaron. De verdad que fueron valiosos y aprendí nuevas cosas que desconocía intentando resolver esto.

## Mi problema fue el antivirus y firewall, algo que definitivamente no pensé.
![image](https://user-images.githubusercontent.com/84040594/161366640-707f1345-e136-4783-891a-fbb4164bc24f.png)

Mis archivos fueron correctamente creados e incluso se agregó el de yarn que instalé. 

De nuevo muchas gracias Explores 🚀🚀🚀🚀🤗🤗

**Reply by gumodi35** [reply]
igual ando atrasado en esa práctica y estoy buscando hacerla en grupo si quieres y puedes colaborar me gustaría hacerla en equipo, estoy en contacto si te interesa.

**Comment by DanyVeneno** [comment]
Bien que bueno saludos
